src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,58 +1,106 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
-
-  return (
-    <form className="NewMovie" key={count}>
-      <h2 className="title">Add a movie</h2>
-
-      <TextField
-        name="title"
-        label="Title"
-        value=""
-        onChange={() => {}}
-        required
-      />
-
-      <TextField
-        name="description"
-        label="Description"
-        value=""
-      />
-
-      <TextField
-        name="imgUrl"
-        label="Image URL"
-        value=""
-      />
-
-      <TextField
-        name="imdbUrl"
-        label="Imdb URL"
-        value=""
-      />
-
-      <TextField
-        name="imdbId"
-        label="Imdb ID"
-        value=""
-      />
-
-      <div className="field is-grouped">
-        <div className="control">
-          <button
-            type="submit"
-            data-cy="submit-button"
-            className="button is-link"
-          >
-            Add
-          </button>
-        </div>
-      </div>
-    </form>
-  );
+const defaultFormValues = {
+  title: '',
+  description: '',
+  imgUrl: '',
+  imdbUrl: '',
+  imdbId: '',
 };
+
+export const NewMovie
+  = ({ onAdd }: { onAdd: (movie: Movie) => void }) => {
+    const [count, setCount] = useState(0);
+    const [formValues, setFormValues] = useState(defaultFormValues);
+
+    const formReset = () => {
+      setFormValues(defaultFormValues);
+    };
+
+    const isAddButtonDisabled = () => {
+      const {
+        title,
+        imgUrl,
+        imdbUrl,
+        imdbId,
+      } = formValues;
+
+      return !(title.trim() && imgUrl.trim()
+        && imdbUrl.trim() && imdbId.trim());
+    };
+
+    const handleSubmit = (): void => {
+      onAdd(formValues);
+
+      formReset();
+
+      setCount((prevCount) => prevCount + 1);
+    };
+
+    const handleChange = (name: string, newValue: string) => {
```

Comment:
The name should clearly describe what is happening in the handler:
```suggestion
    const handleInputChange = (name: string, newValue: string) => {
```